ontology/obo/tree.go

```diff
@@ -0,0 +1,160 @@
+package obo
+
+import (
+	"fmt"
+	"github.com/vertgenlab/gonomics/exception"
+	"github.com/vertgenlab/gonomics/fileio"
+	"log"
+)
+
+// BuildTree takes a slice of Obo structs and builds a tree, setting
+// the Parents and Children fields to point to related nodes.
+func BuildTree(terms []Obo) map[string]*Obo {
```

Comment:
I think there is a tension here where it is unclear if we are treating the obo as a tree, slice, or map.  We can get many slice benefits from a map (e.g. looping through keys), so maybe the question is whether it is a tree or a map.  From the code I am seeing, I think we want the reader to get a root (or set of roots) at the end and treat the ontology as a tree.